### PR TITLE
feat(bigquery): Add examples for creating a table

### PR DESCRIPTION
**Pull request by ccarpentiere**
## Description

Fixes #269181465

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [X] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [X] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [X] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [X] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [ ] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [X] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [X] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s):

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

File: bigquery/bigquery_create_table_iam/main.tf [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+]
+EOF
+
+}
+
+data "google_iam_policy" "default" {
+  binding {
+    role = "roles/bigquery.dataOwner"
+    members = [
+      "user:raha@altostrat.com",
```
Comment (betsy-lichtenberg) [inline bigquery/bigquery_create_table_iam/main.tf:58]: Is this ok to use in public-facing content?

**Comment by ccarpentiere** [comment]
Yes, it is one of the approved names in our style guide.

File: bigquery/bigquery_create_table_iam/main.tf (hunk 1 above)
Comment (ccarpentiere) [reply bigquery/bigquery_create_table_iam/main.tf:58]: Yes, it is one of the approved names in our style guide.

File: bigquery/bigquery_create_table/main.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  location                        = "US"
+  max_time_travel_hours           = 96 # 4 days
+
+  labels = {
+    billing_group = "accounting",
+    pii           = "sensitive"
+  }
+}
+
+resource "google_bigquery_table" "default" {
```
Comment (betsy-lichtenberg) [inline bigquery/bigquery_create_table/main.tf:32]: Please add:

`deletion_protection=false`

with a comment to set it to `true` in production.

Here is the logged error in the test run:

```
Step #5 - "test-samples-3": Error: cannot destroy instance without setting deletion_protection=false and running `terraform apply`
Step #5 - "test-samples-3": }
Step #5 - "test-samples-3":     destroy.go:11: 
Step #5 - "test-samples-3":         	Error Trace:	/workspace/test/integration/destroy.go:11
Step #5 - "test-samples-3":         	            				/workspace/test/integration/terraform.go:350
Step #5 - "test-samples-3":         	            				/workspace/test/integration/terraform.go:413
Step #5 - "test-samples-3":         	            				/workspace/test/integration/terraform.go:426
Step #5 - "test-samples-3":         	            				/workspace/test/integration/stages.go:31
Step #5 - "test-samples-3":         	            				/workspace/test/integration/terraform.go:429
Step #5 - "test-samples-3":         	Error:      	Received unexpected error:
Step #5 - "test-samples-3":         	            	FatalError{Underlying: error while running command: exit status 1; 
Step #5 - "test-samples-3":         	            	Error: cannot destroy instance without setting deletion_protection=false and running `terraform apply`
Step #5 - "test-samples-3":         	            	}
Step #5 - "test-samples-3":         	Test:       	TestSamples/3/bigquery_create_table
```

**Comment by ccarpentiere** [comment]
Is there anything else you need from me on this PR?

**Comment by msampathkumar** [comment]
Updated the PR summary. API is already enabled for BigQuery. No further addition is required.

PR looks good. All requested changes seem to be resolved. To approve the PR, I am waiting for the tests to pass.

**Review by msampathkumar — APPROVED** [review]
All tests are passing. LGTM.

**Pull request merged**